src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6362,6 +6366,13 @@ private static BoundExpression CreatePlaceholderIfNecessary(BoundExpression expr
             // Gets the instance or extension invocation receiver if any.
             bool tryGetReceiver(BoundCall node, [MaybeNullWhen(returnValue: false)] out BoundCall receiver)
             {
+                if (node.Method.GetIsNewExtensionMember())
+                {
+                    // The receiver will be analyzed as an argument instead
+                    receiver = null;
+                    return false;
```

Comment:
>return false; [](http://example.com/codeflow?start=20&length=13)

This defeats the purpose of the method to enable handling of long chains of fluent calls. Note that for classic extension methods we return true (line 6388).